File: apps/api-kotlin/src/main/kotlin/com/couplecalendar/application/dto/response/UserResponse.kt
package com.couplecalendar.application.dto.response

import com.couplecalendar.domain.aggregate.User

data class UserResponse(
    val id: String,
    val email: String,
    val name: String,
    val nickname: String,
    val birthday: String?,
    val bio: String?,
    val partnerNickname: String?,
    val coupleId: String?,
    val createdAt: String,
    val updatedAt: String
) {
    companion object {
        fun fromAggregate(user: User): UserResponse = UserResponse(
            id = user.id.toString(),
            email = user.email.value,
            name = user.name,
            nickname = user.nickname,
            birthday = user.birthday?.toString(),
            bio = user.bio,
            partnerNickname = user.partnerNickname,
            coupleId = user.coupleId?.toString(),
            createdAt = user.createdAt.toString(),
            updatedAt = user.updatedAt.toString()
        )
    }
}

File: apps/web/src/data/dto/user-response.ts
export interface UserResponse {
	id: string;
	email: string;
	/** 실명/표시 이름. 커플 화면에서 노출된다. */
	name: string;
	/** 상대방에게 보이는 닉네임. */
	nickname: string;
	birthday: string | null;
	/** 자기소개. */
	bio: string | null;
	/** 내가 파트너를 부르는 사설 호칭(나에게만 보임). 본인 레코드에만 의미가 있다. */
	partnerNickname: string | null;
	coupleId: string | null;
	createdAt: string;
	updatedAt: string;
}
